libraries/Microsoft.Bot.StreamingExtensions/Payloads/Assemblers/ContentStreamAssembler.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Bot.StreamingExtensions.Payloads
+{
+    internal class ContentStreamAssembler : PayloadAssembler
+    {
+        private readonly IStreamManager _streamManager;
+
+        public ContentStreamAssembler(IStreamManager streamManager, Guid id)
+            : base(id)
+        {
+            ContentType = null;
+            ContentLength = null;
+            _streamManager = streamManager;
+        }
+
+        public ContentStreamAssembler(IStreamManager streamManager, Guid id, string type, int? length)
+            : this(streamManager, id)
+        {
+            ContentType = type;
+            ContentLength = length;
+        }
+
+        public int? ContentLength { get; set; }
+
+        public string ContentType { get; set; }
+
+        public override Stream CreatePayloadStream()
+        {
+            return new PayloadStream(this);
+        }
```

Comment:
Let me see if I understand this correctly: 

A ContentStreamAssemblyer is creating a PayloadStream, but NOT a ContentStream. Is that right? 

A Content Stream == Payload Stream? That doesn't seem correct.